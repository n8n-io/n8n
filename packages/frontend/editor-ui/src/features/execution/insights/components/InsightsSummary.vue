<script setup lang="ts">
import { useTelemetry } from '@/composables/useTelemetry';
import { VIEWS } from '@/constants';
import {
	INSIGHT_IMPACT_TYPES,
	INSIGHTS_UNIT_IMPACT_MAPPING,
} from '@/features/execution/insights/insights.constants';
import type { InsightsSummaryDisplay } from '@/features/execution/insights/insights.types';
import type { InsightsDateRange, InsightsSummary } from '@n8n/api-types';
import { useI18n } from '@n8n/i18n';
import { smartDecimal } from '@n8n/utils/number/smartDecimal';
import { computed, useCssModule } from 'vue';
import { I18nT } from 'vue-i18n';
import { useRoute } from 'vue-router';
import { getTimeRangeLabels } from '../insights.utils';

import { N8nIcon, N8nTooltip } from '@n8n/design-system';
const props = defineProps<{
	summary: InsightsSummaryDisplay;
	timeRange: InsightsDateRange['key'];
	loading?: boolean;
}>();

const i18n = useI18n();
const route = useRoute();
const $style = useCssModule();
const telemetry = useTelemetry();

const timeRangeLabels = getTimeRangeLabels();

const summaryTitles = computed<Record<keyof InsightsSummary, string>>(() => ({
	total: i18n.baseText('insights.banner.title.total'),
	failed: i18n.baseText('insights.banner.title.failed'),
	failureRate: i18n.baseText('insights.banner.title.failureRate'),
	timeSaved: i18n.baseText('insights.banner.title.timeSaved'),
	averageRunTime: i18n.baseText('insights.banner.title.averageRunTime'),
}));

const summaryHasNoData = computed(() => {
	const summaryValues = Object.values(props.summary);
	return summaryValues.length > 0 && summaryValues.every((summary) => !summary.value);
});

const summaryWithRouteLocations = computed(() =>
	props.summary.map((s) => ({
		...s,
		to: { name: VIEWS.INSIGHTS, params: { insightType: s.id }, query: route.query },
	})),
);

const getImpactStyle = (id: keyof InsightsSummary, value: number) => {
	const impact = INSIGHTS_UNIT_IMPACT_MAPPING[id];
	if (value === 0 || impact === INSIGHT_IMPACT_TYPES.NEUTRAL) {
		return $style.neutral;
	}
	if (impact === INSIGHT_IMPACT_TYPES.POSITIVE) {
		return value > 0 ? $style.positive : $style.negative;
	}
	if (impact === INSIGHT_IMPACT_TYPES.NEGATIVE) {
		return value < 0 ? $style.positive : $style.negative;
	}
	return $style.neutral;
};

const trackTabClick = (insightType: keyof InsightsSummary) => {
	telemetry.track(`User clicked ${summaryTitles.value[insightType]}`, {
		referrer: route.name === VIEWS.INSIGHTS ? 'Dashboard' : 'Overview',
	});
};
</script>

<template>
	<div :class="$style.insights">
		<ul data-test-id="insights-summary-tabs">
			<li
				v-for="{ id, value, deviation, deviationUnit, unit, to } in summaryWithRouteLocations"
				:key="id"
				:data-test-id="`insights-summary-tab-${id}`"
			>
				<N8nTooltip
					:placement="route.name === VIEWS.INSIGHTS ? 'bottom' : 'top'"
					:disabled="!(summaryHasNoData && id === 'total')"
					:show-after="500"
				>
					<template #content>
						<I18nT keypath="insights.banner.noData.tooltip" scope="global">
							<template #link>
								<a :href="i18n.baseText('insights.banner.noData.tooltip.link.url')" target="_blank">
									{{ i18n.baseText('insights.banner.noData.tooltip.link') }}
								</a>
							</template>
						</I18nT>
					</template>
					<RouterLink :to="to" :exact-active-class="$style.activeTab" @click="trackTabClick(id)">
						<strong>
							<N8nTooltip placement="bottom" :disabled="id !== 'timeSaved'">
								<template #content>
									{{ i18n.baseText('insights.banner.title.timeSaved.tooltip') }}
								</template>
								{{ summaryTitles[id] }}
							</N8nTooltip>
						</strong>
						<small :class="$style.days">
							{{ timeRangeLabels[timeRange] }}
						</small>
						<span v-if="value === 0 && id === 'timeSaved'" :class="$style.empty">
							<em>--</em>
							<small>
								<N8nTooltip placement="bottom">
									<template #content>
										<I18nT keypath="insights.banner.timeSaved.tooltip" scope="global">
											<template #link>{{
												i18n.baseText('insights.banner.timeSaved.tooltip.link.text')
											}}</template>
										</I18nT>
									</template>
									<N8nIcon :class="$style.icon" icon="info" size="medium" />
								</N8nTooltip>
							</small>
						</span>
						<span v-else>
							<em
								>{{ smartDecimal(value).toLocaleString('en-US') }} <i>{{ unit }}</i></em
							>
							<small v-if="deviation !== null" :class="getImpactStyle(id, deviation)">
								<N8nIcon
									:class="[$style.icon, getImpactStyle(id, deviation)]"
									:icon="
										deviation === 0
											? 'chevron-right'
											: deviation > 0
												? 'chevron-up'
												: 'chevron-down'
									"
								/>
								<N8nTooltip placement="bottom" :disabled="id !== 'failureRate'">
									<template #content>
										{{ i18n.baseText('insights.banner.failureRate.deviation.tooltip') }}
									</template>
									{{ smartDecimal(Math.abs(deviation)).toLocaleString('en-US') }}{{ deviationUnit }}
								</N8nTooltip>
							</small>
						</span>
					</RouterLink>
				</N8nTooltip>
			</li>
		</ul>
	</div>
</template>

<style lang="scss" module>
.insights {
	display: grid;
	grid-template-rows: auto 1fr;
	padding: var(--spacing--xs) 0 var(--spacing--2xl);

	ul {
		display: flex;
		height: 101px;
		align-items: stretch;
		justify-content: space-evenly;
		border: var(--border-width) var(--border-style) var(--color--foreground);
		border-radius: 6px;
		list-style: none;
		overflow-x: auto;

		li {
			display: flex;
			justify-content: stretch;
			align-items: stretch;
			flex: 1 0;
			border-left: var(--border-width) var(--border-style) var(--color--foreground);

			&:first-child {
				border-left: 0;
			}
		}

		a {
			display: grid;
			align-items: center;
			align-content: center;
			width: 100%;
			height: 100%;
			padding: var(--spacing--3xs) var(--spacing--lg) 0;
			border-bottom: 3px solid transparent;

			&:hover {
				background-color: var(--color--background--light-3);
				transition: background-color 0.3s;
			}

			&.activeTab {
				background-color: var(--color--background--light-3);
				border-color: var(--color--primary);
				transition: background-color 0.3s ease-in-out;
			}

			strong {
				justify-self: flex-start;
				color: var(--color--text--shade-1);
				font-size: var(--font-size--sm);
				font-weight: 400;
				white-space: nowrap;
				margin-bottom: var(--spacing--3xs);
			}

			.days {
				padding: 0;
				margin: 0 0 var(--spacing--xs);
				color: var(--color--text--tint-1);
				font-size: var(--font-size--2xs);
				font-weight: var(--font-weight-normal);
			}

			span {
				display: flex;
				align-items: baseline;

				&.empty {
					em {
						color: var(--color--text--tint-2);
						body[data-theme='dark'] & {
							color: var(--color--text--tint-1);
						}
					}
					small {
						padding: 0;
						height: 21px;
						font-weight: var(--font-weight--bold);

						.icon {
							top: 5px;
							transform: translateY(0);
							color: var(--color--text--tint-1);
						}
					}
				}
			}

			em {
				display: flex;
				align-items: baseline;
				justify-content: flex-start;
				color: var(--color--text--shade-1);
				font-size: 24px;
				line-height: 100%;
				font-weight: 600;
				font-style: normal;
				gap: var(--spacing--5xs);

				i {
					font-size: 22px;
					font-style: normal;
				}
			}

			small {
				position: relative;
				display: flex;
				align-items: center;
				padding: 0 0 0 14px;
				margin: 0 0 0 var(--spacing--xs);
				font-size: var(--font-size--2xs);
				font-weight: var(--font-weight--bold);
				white-space: nowrap;
			}
		}
	}

	.noData {
		em {
			color: var(--color--text--tint-1);
			font-size: var(--font-size--md);
		}
	}
}

.positive {
	color: var(--color--success);
}

.negative {
	color: var(--color--danger);
}

.neutral {
	color: var(--color--text--tint-1);

	.icon {
		font-size: 17px;
	}
}

.icon {
	position: absolute;
	font-size: 17px;
	left: 0;
	top: 50%;
	transform: translateY(-50%);
}

.loading {
	display: flex;
	align-self: stretch;
	align-items: stretch;

	> div {
		margin: 0;
		height: auto;
		border-radius: inherit;
	}
}
</style>
