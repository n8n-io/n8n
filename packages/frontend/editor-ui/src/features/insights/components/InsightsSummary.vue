<script setup lang="ts">
import { computed, useCssModule } from 'vue';
import { smartDecimal } from '@n8n/utils/number/smartDecimal';
import { useI18n } from '@/composables/useI18n';
import type { InsightsSummary } from '@n8n/api-types';
import type { InsightsSummaryDisplay } from '@/features/insights/insights.types';
import {
	INSIGHT_IMPACT_TYPES,
	INSIGHTS_UNIT_IMPACT_MAPPING,
} from '@/features/insights/insights.constants';

defineProps<{
	summary: InsightsSummaryDisplay;
	loading?: boolean;
}>();

const i18n = useI18n();
const $style = useCssModule();

const summaryTitles = computed<Record<keyof InsightsSummary, string>>(() => ({
	total: i18n.baseText('insights.banner.title.total'),
	failed: i18n.baseText('insights.banner.title.failed'),
	failureRate: i18n.baseText('insights.banner.title.failureRate'),
	timeSaved: i18n.baseText('insights.banner.title.timeSaved'),
	averageRunTime: i18n.baseText('insights.banner.title.averageRunTime'),
}));

const getSign = (n: number) => (n > 0 ? '+' : undefined);
const getImpactStyle = (id: keyof InsightsSummary, value: number) => {
	const impact = INSIGHTS_UNIT_IMPACT_MAPPING[id];
	if (value === 0 || impact === INSIGHT_IMPACT_TYPES.NEUTRAL) {
		return $style.neutral;
	}
	if (impact === INSIGHT_IMPACT_TYPES.POSITIVE) {
		return value > 0 ? $style.positive : $style.negative;
	}
	if (impact === INSIGHT_IMPACT_TYPES.NEGATIVE) {
		return value < 0 ? $style.positive : $style.negative;
	}
	return $style.neutral;
};
</script>

<template>
	<div v-if="summary.length" :class="$style.insights">
		<N8nHeading bold tag="h3" size="small" color="text-light" class="mb-xs">{{
			i18n.baseText('insights.banner.title', { interpolate: { count: 7 } })
		}}</N8nHeading>
		<N8nLoading v-if="loading" :class="$style.loading" :cols="5" />
		<ul v-else data-test-id="insights-summary-tabs">
			<li
				v-for="{ id, value, deviation, unit } in summary"
				:key="id"
				:data-test-id="`insights-summary-tab-${id}`"
			>
				<p>
					<strong>{{ summaryTitles[id] }}</strong>
					<span v-if="value === 0 && id === 'timeSaved'" :class="$style.empty">
						<em>--</em>
						<small>
							<N8nTooltip placement="bottom">
								<template #content>
									<i18n-t keypath="insights.banner.timeSaved.tooltip">
										<template #link>
											<a href="#">{{
												i18n.baseText('insights.banner.timeSaved.tooltip.link.text')
											}}</a>
										</template>
									</i18n-t>
								</template>
								<N8nIcon :class="$style.icon" icon="info-circle" />
							</N8nTooltip>
						</small>
					</span>
					<span v-else>
						<em
							>{{ smartDecimal(value) }} <i>{{ unit }}</i></em
						>
						<small :class="getImpactStyle(id, deviation)">
							<N8nIcon
								:class="[$style.icon, getImpactStyle(id, deviation)]"
								:icon="deviation === 0 ? 'caret-right' : deviation > 0 ? 'caret-up' : 'caret-down'"
							/>
							{{ getSign(deviation) }}{{ smartDecimal(deviation) }}
						</small>
					</span>
				</p>
			</li>
		</ul>
	</div>
</template>

<style lang="scss" module>
.insights {
	display: grid;
	grid-template-rows: auto 1fr;
	padding: var(--spacing-xs) 0 var(--spacing-2xl);

	ul {
		display: flex;
		height: 91px;
		align-items: stretch;
		justify-content: flex-start;
		border: var(--border-width-base) var(--border-style-base) var(--color-foreground-base);
		border-radius: 6px;
		list-style: none;
		background-color: var(--color-background-xlight);
		overflow-x: auto;

		li {
			display: flex;
			justify-content: flex-start;
			align-items: center;
			flex: 1;
			border-left: var(--border-width-base) var(--border-style-base) var(--color-foreground-base);
			padding: 0 var(--spacing-xl) 0 var(--spacing-l);

			&:first-child {
				border-left: 0;
			}
		}

		p {
			display: grid;

			strong {
				color: var(--color-text-dark);
				font-size: var(--font-size-s);
				font-weight: 400;
				white-space: nowrap;
				margin-bottom: var(--spacing-2xs);
			}

			span {
				display: flex;
				align-items: baseline;
				gap: var(--spacing-xs);

				&.empty {
					em {
						color: var(--color-text-lighter);
					}
					small {
						padding: 0;
						height: 21px;
						font-weight: var(--font-weight-bold);

						.icon {
							height: 20px;
							width: 8px;
							top: -3px;
							transform: translateY(0);
							color: var(--color-text-light);
						}
					}
				}
			}

			em {
				display: flex;
				align-items: baseline;
				justify-content: flex-start;
				color: var(--color-text-dark);
				font-size: var(--font-size-2xl);
				line-height: 100%;
				font-weight: 600;
				font-style: normal;
				gap: var(--spacing-5xs);

				i {
					color: var(--color-text-light);
					font-size: 22px;
					font-style: normal;
				}
			}

			small {
				position: relative;
				display: flex;
				align-items: center;
				padding: 0 0 0 18px;
				font-size: 14px;
				font-weight: 400;
				white-space: nowrap;
			}
		}
	}
}

.positive {
	color: var(--color-success);
}

.negative {
	color: var(--color-danger);
}

.neutral {
	color: var(--color-text-light);

	.icon {
		font-size: 23px;
	}
}

.icon {
	position: absolute;
	font-size: 32px;
	left: 0;
	top: 50%;
	transform: translateY(-50%);
}

.loading {
	display: flex;
	align-self: stretch;
	align-items: stretch;

	> div {
		margin: 0;
		height: auto;
	}
}
</style>
