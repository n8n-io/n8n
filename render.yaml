services:
  - type: web
    name: luci-n8n-agent
    env: docker
    plan: free
    branch: master
    autoDeploy: true
    buildCommand: ""
    startCommand: "n8n start"
    envVars:
      - key: N8N_BASIC_AUTH_ACTIVE
        value: "true"
      - key: N8N_BASIC_AUTH_USER
        value: "admin"
      - key: N8N_BASIC_AUTH_PASSWORD
        value: "admin"
      - key: N8N_HOST
        value: "0.0.0.0"
      - key: N8N_PORT
        value: "5678"
      - key: WEBHOOK_URL
        value: https://luci-n8n-agent.onrender.com/
