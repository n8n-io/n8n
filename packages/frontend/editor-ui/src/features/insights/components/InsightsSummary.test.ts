import InsightsSummary from '@/features/insights/components/InsightsSummary.vue';
import { createComponentRenderer } from '@/__tests__/render';
import type { InsightsSummaryDisplay } from '@/features/insights/insights.types';

const renderComponent = createComponentRenderer(InsightsSummary);

describe('InsightsSummary', () => {
	it('should render without error', () => {
		expect(() =>
			renderComponent({
				props: {
					summary: [],
				},
			}),
		).not.toThrow();
	});

	test.each<InsightsSummaryDisplay[]>([
		[[]],
		[
			[
				{ id: 'total', value: 525, deviation: 85, unit: '' },
				{ id: 'failed', value: 14, deviation: 3, unit: '' },
				{ id: 'failureRate', value: 1.9, deviation: -0.8, unit: '%' },
				{ id: 'timeSaved', value: 55.55555555555556, deviation: -5.164722222222222, unit: 'h' },
				{ id: 'averageRunTime', value: 2.5, deviation: -0.5, unit: 's' },
			],
		],
		[
			[
				{ id: 'total', value: 525, deviation: 85, unit: '' },
				{ id: 'failed', value: 14, deviation: 3, unit: '' },
				{ id: 'failureRate', value: 1.9, deviation: -0.8, unit: '%' },
				{ id: 'timeSaved', value: 0, deviation: 0, unit: 'h' },
				{ id: 'averageRunTime', value: 2.5, deviation: -0.5, unit: 's' },
			],
		],
		[
			[
				{ id: 'total', value: 525, deviation: -2, unit: '' },
				{ id: 'failed', value: 14, deviation: -3, unit: '' },
				{ id: 'failureRate', value: 1.9, deviation: 0.8, unit: '%' },
				{ id: 'timeSaved', value: 55.55555555555556, deviation: 0, unit: 'h' },
				{ id: 'averageRunTime', value: 2.5, deviation: 0.5, unit: 's' },
			],
		],
	])('should render the summary correctly', (summary) => {
		const { html } = renderComponent({
			props: {
				summary,
			},
		});

		expect(html()).toMatchSnapshot();
	});
});
