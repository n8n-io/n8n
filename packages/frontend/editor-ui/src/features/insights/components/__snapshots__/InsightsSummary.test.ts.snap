// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InsightsSummary > should render the summary correctly 1`] = `"<!--v-if-->"`;

exports[`InsightsSummary > should render the summary correctly 2`] = `
"<div class="insights">
  <h3 class="n8n-heading text-light size-small bold mb-xs mb-xs">Production executions for the last 7 days</h3>
  <ul data-test-id="insights-summary-tabs">
    <li data-test-id="insights-summary-tab-total">
      <p><strong>Total</strong><span><em>525 <i></i></em><small class="positive"><span class="n8n-text compact size-medium regular n8n-icon icon positive icon positive n8n-icon icon positive icon positive"><svg class="svg-inline--fa fa-caret-up fa-w-10 medium" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="caret-up" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path class="" fill="currentColor" d="M288.662 352H31.338c-17.818 0-26.741-21.543-14.142-34.142l128.662-128.662c7.81-7.81 20.474-7.81 28.284 0l128.662 128.662c12.6 12.599 3.676 34.142-14.142 34.142z"></path></svg></span> +85</small></span></p>
    </li>
    <li data-test-id="insights-summary-tab-failed">
      <p><strong>Failed</strong><span><em>14 <i></i></em><small class="negative"><span class="n8n-text compact size-medium regular n8n-icon icon negative icon negative n8n-icon icon negative icon negative"><svg class="svg-inline--fa fa-caret-up fa-w-10 medium" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="caret-up" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path class="" fill="currentColor" d="M288.662 352H31.338c-17.818 0-26.741-21.543-14.142-34.142l128.662-128.662c7.81-7.81 20.474-7.81 28.284 0l128.662 128.662c12.6 12.599 3.676 34.142-14.142 34.142z"></path></svg></span> +3</small></span></p>
    </li>
    <li data-test-id="insights-summary-tab-failureRate">
      <p><strong>Failure rate</strong><span><em>1.9 <i>%</i></em><small class="positive"><span class="n8n-text compact size-medium regular n8n-icon icon positive icon positive n8n-icon icon positive icon positive"><svg class="svg-inline--fa fa-caret-down fa-w-10 medium" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="caret-down" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path class="" fill="currentColor" d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"></path></svg></span> -0.8</small></span></p>
    </li>
    <li data-test-id="insights-summary-tab-timeSaved">
      <p><strong>Time saved</strong><span><em>55.56 <i>h</i></em><small class="negative"><span class="n8n-text compact size-medium regular n8n-icon icon negative icon negative n8n-icon icon negative icon negative"><svg class="svg-inline--fa fa-caret-down fa-w-10 medium" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="caret-down" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path class="" fill="currentColor" d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"></path></svg></span> -5.16</small></span></p>
    </li>
    <li data-test-id="insights-summary-tab-averageRunTime">
      <p><strong>Avg. run time</strong><span><em>2.5 <i>s</i></em><small class="neutral"><span class="n8n-text compact size-medium regular n8n-icon icon neutral icon neutral n8n-icon icon neutral icon neutral"><svg class="svg-inline--fa fa-caret-down fa-w-10 medium" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="caret-down" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path class="" fill="currentColor" d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"></path></svg></span> -0.5</small></span></p>
    </li>
  </ul>
</div>"
`;

exports[`InsightsSummary > should render the summary correctly 3`] = `
"<div class="insights">
  <h3 class="n8n-heading text-light size-small bold mb-xs mb-xs">Production executions for the last 7 days</h3>
  <ul data-test-id="insights-summary-tabs">
    <li data-test-id="insights-summary-tab-total">
      <p><strong>Total</strong><span><em>525 <i></i></em><small class="positive"><span class="n8n-text compact size-medium regular n8n-icon icon positive icon positive n8n-icon icon positive icon positive"><svg class="svg-inline--fa fa-caret-up fa-w-10 medium" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="caret-up" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path class="" fill="currentColor" d="M288.662 352H31.338c-17.818 0-26.741-21.543-14.142-34.142l128.662-128.662c7.81-7.81 20.474-7.81 28.284 0l128.662 128.662c12.6 12.599 3.676 34.142-14.142 34.142z"></path></svg></span> +85</small></span></p>
    </li>
    <li data-test-id="insights-summary-tab-failed">
      <p><strong>Failed</strong><span><em>14 <i></i></em><small class="negative"><span class="n8n-text compact size-medium regular n8n-icon icon negative icon negative n8n-icon icon negative icon negative"><svg class="svg-inline--fa fa-caret-up fa-w-10 medium" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="caret-up" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path class="" fill="currentColor" d="M288.662 352H31.338c-17.818 0-26.741-21.543-14.142-34.142l128.662-128.662c7.81-7.81 20.474-7.81 28.284 0l128.662 128.662c12.6 12.599 3.676 34.142-14.142 34.142z"></path></svg></span> +3</small></span></p>
    </li>
    <li data-test-id="insights-summary-tab-failureRate">
      <p><strong>Failure rate</strong><span><em>1.9 <i>%</i></em><small class="positive"><span class="n8n-text compact size-medium regular n8n-icon icon positive icon positive n8n-icon icon positive icon positive"><svg class="svg-inline--fa fa-caret-down fa-w-10 medium" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="caret-down" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path class="" fill="currentColor" d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"></path></svg></span> -0.8</small></span></p>
    </li>
    <li data-test-id="insights-summary-tab-timeSaved">
      <p><strong>Time saved</strong><span class="empty"><em>--</em><small><span class="n8n-text compact size-medium regular n8n-icon icon el-tooltip__trigger el-tooltip__trigger icon el-tooltip__trigger el-tooltip__trigger n8n-icon icon el-tooltip__trigger el-tooltip__trigger icon el-tooltip__trigger el-tooltip__trigger"><svg class="svg-inline--fa fa-info-circle fa-w-16 medium" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="info-circle" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path class="" fill="currentColor" d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"></path></svg></span>
        <!--teleport start-->
        <!--teleport end--></small></span>
      </p>
    </li>
    <li data-test-id="insights-summary-tab-averageRunTime">
      <p><strong>Avg. run time</strong><span><em>2.5 <i>s</i></em><small class="neutral"><span class="n8n-text compact size-medium regular n8n-icon icon neutral icon neutral n8n-icon icon neutral icon neutral"><svg class="svg-inline--fa fa-caret-down fa-w-10 medium" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="caret-down" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path class="" fill="currentColor" d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"></path></svg></span> -0.5</small></span></p>
    </li>
  </ul>
</div>"
`;

exports[`InsightsSummary > should render the summary correctly 4`] = `
"<div class="insights">
  <h3 class="n8n-heading text-light size-small bold mb-xs mb-xs">Production executions for the last 7 days</h3>
  <ul data-test-id="insights-summary-tabs">
    <li data-test-id="insights-summary-tab-total">
      <p><strong>Total</strong><span><em>525 <i></i></em><small class="negative"><span class="n8n-text compact size-medium regular n8n-icon icon negative icon negative n8n-icon icon negative icon negative"><svg class="svg-inline--fa fa-caret-down fa-w-10 medium" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="caret-down" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path class="" fill="currentColor" d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"></path></svg></span> -2</small></span></p>
    </li>
    <li data-test-id="insights-summary-tab-failed">
      <p><strong>Failed</strong><span><em>14 <i></i></em><small class="positive"><span class="n8n-text compact size-medium regular n8n-icon icon positive icon positive n8n-icon icon positive icon positive"><svg class="svg-inline--fa fa-caret-down fa-w-10 medium" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="caret-down" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path class="" fill="currentColor" d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"></path></svg></span> -3</small></span></p>
    </li>
    <li data-test-id="insights-summary-tab-failureRate">
      <p><strong>Failure rate</strong><span><em>1.9 <i>%</i></em><small class="negative"><span class="n8n-text compact size-medium regular n8n-icon icon negative icon negative n8n-icon icon negative icon negative"><svg class="svg-inline--fa fa-caret-up fa-w-10 medium" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="caret-up" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path class="" fill="currentColor" d="M288.662 352H31.338c-17.818 0-26.741-21.543-14.142-34.142l128.662-128.662c7.81-7.81 20.474-7.81 28.284 0l128.662 128.662c12.6 12.599 3.676 34.142-14.142 34.142z"></path></svg></span> +0.8</small></span></p>
    </li>
    <li data-test-id="insights-summary-tab-timeSaved">
      <p><strong>Time saved</strong><span><em>55.56 <i>h</i></em><small class="neutral"><span class="n8n-text compact size-medium regular n8n-icon icon neutral icon neutral n8n-icon icon neutral icon neutral"><svg class="svg-inline--fa fa-caret-right fa-w-6 medium" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="caret-right" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 192 512"><path class="" fill="currentColor" d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"></path></svg></span> 0</small></span></p>
    </li>
    <li data-test-id="insights-summary-tab-averageRunTime">
      <p><strong>Avg. run time</strong><span><em>2.5 <i>s</i></em><small class="neutral"><span class="n8n-text compact size-medium regular n8n-icon icon neutral icon neutral n8n-icon icon neutral icon neutral"><svg class="svg-inline--fa fa-caret-up fa-w-10 medium" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="caret-up" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path class="" fill="currentColor" d="M288.662 352H31.338c-17.818 0-26.741-21.543-14.142-34.142l128.662-128.662c7.81-7.81 20.474-7.81 28.284 0l128.662 128.662c12.6 12.599 3.676 34.142-14.142 34.142z"></path></svg></span> +0.5</small></span></p>
    </li>
  </ul>
</div>"
`;
